addons/google_calendar/static/src/views/google_calendar/google_calendar_model.js

```diff
@@ -24,10 +24,7 @@ patch(AttendeeCalendarModel.prototype, "google_calendar_google_calendar_model_fu
             return _super(...arguments);
         }
         try {
-            await Promise.race([
-                new Promise(resolve => setTimeout(resolve, 1000)),
-                this.syncGoogleCalendar(true)
-            ]);
+            await this.syncGoogleCalendar(true);
```

Comment:
I assume that the former code was there to avoid freezing the view while syncing google/microsoft events. I suggest to keep it, and instead checking that the component is still alive before calling super:

```js
// on top
import { status } from "@odoo/owl";
// here
if (status(this) !== "destroyed") {
    return _super(...arguments);
}
```

Reply:
Also, this should be tested (qunit)

Reply:
Hello @aab-odoo,

Sorry for the late reply. I'm encountering an issue, It's giving an error. while debugging, I'm not getting `__owl__` in the component (AttendeeCalendarModel), so the status is also undefined. Do you have any suggestions or am I making some mistake doing this 🤔

Thank you.

Reply:
Damn of course it doesn't work, you're not in a component. My bad. I'm not a fan of this, but I don't see any better solution.
```diff
diff --git a/addons/google_calendar/static/src/views/google_calendar/google_calendar_model.js b/addons/google_calendar/static/src/views/google_calendar/google_calendar_model.js
index a1025013f556..3233990f428b 100644
--- a/addons/google_calendar/static/src/views/google_calendar/google_calendar_model.js
+++ b/addons/google_calendar/static/src/views/google_calendar/google_calendar_model.js
@@ -12,6 +12,7 @@ patch(AttendeeCalendarModel.prototype, {
     setup(params, { rpc }) {
         super.setup(...arguments);
         this.rpc = rpc;
+        this.isAlive = params.isAlive;
         this.googlePendingSync = false;
         this.state = useState({
             googleIsSync: true,
@@ -38,7 +39,9 @@ patch(AttendeeCalendarModel.prototype, {
             console.error("Could not synchronize Google events now.", error);
             this.googlePendingSync = false;
         }
-        return super.updateData(...arguments);
+        if (this.isAlive()) {
+            return super.updateData(...arguments);
+        }
     },
 
     async syncGoogleCalendar(silent = false) {
diff --git a/addons/web/static/src/model/model.js b/addons/web/static/src/model/model.js
index 745a2fd2cb7e..dd78edca314f 100644
--- a/addons/web/static/src/model/model.js
+++ b/addons/web/static/src/model/model.js
@@ -5,7 +5,7 @@ import { SEARCH_KEYS } from "@web/search/with_search/with_search";
 import { useSetupView } from "@web/views/view_hook";
 import { buildSampleORM } from "./sample_server";
 
-import { EventBus, onWillStart, onWillUpdateProps, useComponent } from "@odoo/owl";
+import { EventBus, onWillStart, onWillUpdateProps, status, useComponent } from "@odoo/owl";
 
 /**
  * @typedef {import("@web/search/search_model").SearchParams} SearchParams
@@ -119,7 +119,9 @@ export function useModelWithSampleData(ModelClass, params, options = {}) {
         services[key] = useService(key);
     }
     services.orm = services.orm || useService("orm");
-
+    if (!("isAlive" in params)) {
+        params.isAlive = () => status(component) !== "destroyed";
+    }
     const model = new ModelClass(component.env, params, services);
 
     useBus(
```